### PR TITLE
Create Blaneidx production, and encode lane index as u8

### PR DESCRIPTION
**Pull request by ngzhian**
Also fix up the definition in the syntax to be a u4 instead of a byte.

### CODE REVIEW

**Review by binji — COMMENTED** [review]
I don't see where this is defined as a `u4`...? But also, I don't think we want to change this to a `u4` anyway, without discussion, since that makes a larger lane index a decoding error instead of a validation error.

**Comment by ngzhian** [comment]
Sorry, latest commit changed it to u8, but I didn't update the PR description, will change it now.

**Review by binji — APPROVED** [review]

File: document/core/binary/instructions.rst [hunk 1]
```diff
@@ -442,34 +442,37 @@ The |VCONST| instruction is followed by 16 immediate bytes, which are converted
...
 .. _binary-vternop:
+.. _binary-laneidx:
 
-The |SHUFFLE| instruction is also followed by 16 immediate bytes:
+The |SHUFFLE| instruction is also followed by the encoding of 16 |laneidx| immediates.
 
 .. math::
    \begin{array}{llclll}
+   \production{lane index} & \Blaneidx &::=&
+     l{:}\Bu8 &\Rightarrow& l \\
```
Comment (rossberg) [inline document/core/binary/instructions.rst:452]: Note that Bu8 actually is a LEB128 representation. Here, you'll still need to use Bbyte.

Comment (ngzhian) [reply document/core/binary/instructions.rst:452]: Good catch, fixed in #357

**Pull request merged**